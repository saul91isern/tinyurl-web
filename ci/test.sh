#!/bin/sh

echo "----- Install dependencies -----"
yarn
echo "----- Linter -----"
yarn test --ci

