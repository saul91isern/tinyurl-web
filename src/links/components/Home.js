import React from "react";
import {
  Container,
  Icon,
  Header,
  Grid,
  Menu,
  Segment,
} from "semantic-ui-react";
import Links from "./Links";
import LinkForm from "./LinkForm";

const Home = () => (
  <>
    <Menu attached="top"></Menu>
    <Segment attached="bottom">
      <Grid columns={2} relaxed="very">
        <Grid.Column>
          <Links />
        </Grid.Column>
        <Grid.Column>
          <Container textAlign="center">
            <Header icon>
              <Icon name="world" />
              Add New Url
            </Header>
          </Container>
          <Container>
            <LinkForm />
          </Container>
        </Grid.Column>
      </Grid>
    </Segment>
  </>
);

export default Home;
