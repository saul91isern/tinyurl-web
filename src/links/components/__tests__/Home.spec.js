/* eslint-disable quotes */
import React from "react";
import { render } from "../../../test/render";
import { Home } from "../Home";

describe("<Home />", () => {
  it("renders Home when not loading", async () => {
    const { container, queryByText } = render(<Home loading={false} />, {
      state: { links: [], message: {} },
    });
    expect(queryByText(/My first url shortener/)).toBeTruthy();
    expect(queryByText(/URL Shortener/)).toBeTruthy();
    expect(queryByText(/Add New Url/)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
