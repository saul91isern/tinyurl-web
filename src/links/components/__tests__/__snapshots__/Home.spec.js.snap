// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders Home when not loading 1`] = `
<div>
  <div
    class="ui top attached menu"
  >
    <div
      class="header item"
    >
      <i
        aria-hidden="true"
        class="world big icon"
      />
       My first url shortener
    </div>
  </div>
  <div
    class="ui text container"
    style="margin-top: 40px;"
  />
  <div
    class="ui fluid container"
    style="margin-top: 40px;"
  >
    <div
      class="ui grid"
    >
      <div
        class="row"
      >
        <div
          class="center aligned column"
        >
          <h2
            class="ui header"
          >
            URL Shortener
            <div
              class="sub header"
            >
              Manage and simplify your urls.
            </div>
          </h2>
        </div>
      </div>
    </div>
    <div
      class="ui placeholder segment"
    >
      <div
        class="ui divided two column grid"
      >
        <div
          class="row"
        >
          <div
            class="middle aligned column"
          >
            <div
              class="ui icon input"
            >
              <input
                placeholder="Search..."
                type="text"
                value=""
              />
              <i
                aria-hidden="true"
                class="search circular icon"
              />
            </div>
            <div
              class="ui icon message"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
              <div
                class="content"
              >
                <div
                  class="header"
                >
                  No results found
                </div>
                <p>
                  Maybe you want shorten a new url first.
                </p>
              </div>
            </div>
          </div>
          <div
            class="middle aligned column"
          >
            <div
              class="ui center aligned container"
            >
              <div
                class="ui icon header"
              >
                <i
                  aria-hidden="true"
                  class="world icon"
                />
                Add New Url
              </div>
            </div>
            <div
              class="ui container"
            >
              <form
                class="ui form"
              >
                <div
                  class="required field"
                >
                  <label
                    for="url"
                  >
                    Url
                  </label>
                  <div
                    class="ui fluid input"
                  >
                    <input
                      autocomplete="off"
                      id="url"
                      placeholder="Enter a valid url"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <button
                  class="ui disabled button"
                  disabled=""
                  tabindex="-1"
                  type="submit"
                >
                  Submit
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
